src/Microsoft.DotNet.Tools.Publish/Program.cs

```diff
@@ -50,18 +54,58 @@ public static int Main(string[] args)
                     path = Directory.GetCurrentDirectory();
                 }
 
-                // Load project context and publish it
-                var fx = NuGetFramework.Parse(framework.Value());
-                var rids = new[] { runtime.Value() };
-                var context = ProjectContext.Create(path, fx, rids);
+                // get the lock file
+                var projectDir = path.ToLower().EndsWith(Project.FileName) ?
+                                    Path.GetDirectoryName(path) :
+                                    path;
+
+                var projectLockJsonPath = Path.Combine(projectDir, LockFile.FileName);
+                if (!File.Exists(projectLockJsonPath))
+                {
+                    Reporter.Output.WriteLine($"Unable to locate {LockFile.FileName} for project '{project.Value}'".Red());
+                    Reporter.Output.WriteLine($"Run 'dotnet restore' before calling 'dotnet publish'".Red());
+                    return 1;
+                }
+
+                var lockFile = LockFileReader.Read(projectLockJsonPath);
 
-                if (string.IsNullOrEmpty(context.RuntimeIdentifier))
+                var lockFileTargets = lockFile.Targets.Where(target =>
+                {
+                    if(target.TargetFramework == null || string.IsNullOrEmpty(target.RuntimeIdentifier))
```

Comment:
Please refactor this into an well named method (s). It is not easy to follow the conditions to select the targets as it is.